pkg/metric/metric.go

```diff
@@ -0,0 +1,46 @@
+package metric
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	reqCnt = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "http_requests_total",
+		Help: "The total number of processed requests",
+	}, []string{"method", "host", "path", "status"})
+	reqDur = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "http_request_duration_seconds",
+		Help: "Duration of HTTP requests in seconds",
+	}, []string{"method", "host", "path", "status"})
+	reqSz = promauto.NewSummary(prometheus.SummaryOpts{
+		Name: "http_request_length_bytes",
+		Help: "Content-Length of HTTP requests",
+	})
+	resSz = promauto.NewSummary(prometheus.SummaryOpts{
+		Name: "http_response_length_bytes",
+		Help: "Content-Length of HTTP responses",
+	})
+)
```

Comment:
We're sharing these metrics between plugins. Shouldn't we have different metrics for each plugin, for example, have a `reqCnt` for a server and another for data sources?

Reply:
The metric already has dedicated labels for supporting multiple "hosts" so I guess there's no need to create a different metric. A metric should convey a way of measuring something and labels should differentiate the characteristics of the thing that is being measured ([docs](https://prometheus.io/docs/practices/naming/#labels)).